src/Servers/Kestrel/Transport.Quic/src/Internal/QuicConnectionListener.cs

```diff
@@ -26,6 +26,11 @@ internal class QuicConnectionListener : IMultiplexedConnectionListener, IAsyncDi
 
         public QuicConnectionListener(QuicTransportOptions options, IQuicTrace log, EndPoint endpoint, SslServerAuthenticationOptions sslServerAuthenticationOptions)
         {
+            if (!QuicImplementationProviders.Default.IsSupported)
+            {
+                throw new NotSupportedException("Quic is not supported or enabled on this platform.");
```

Comment:
Worth mentioning the switch in the exception message?

Reply:
The switch implementation is still under active debate. We'll add more detailed guidance when that settles.

Reply:
I think for p7 we should set that flag to true inside asp.net and then check IsSupported.

If we decide to add our own flag we can do that in RC1. @davidfowl

Reply:
We'll see what the runtime team ends up doing with their switch first.

This IsSupported check is still relevant even if the switch is set since it will also check for the dll and other dependencies.